server/src/main/java/com/exacaster/lighter/application/sessions/SessionService.java

```diff
@@ -39,23 +41,32 @@ public SessionService(ApplicationStorage applicationStorage,
         this.statementHandler = statementHandler;
     }
 
+    //TODO do we wanna return just Session or PermSession as well?
     public List<Application> fetch(Integer from, Integer size) {
         return applicationStorage.findApplications(ApplicationType.SESSION, from, size);
     }
 
     public Application createSession(SubmitParams params) {
-        return createSession(params, UUID.randomUUID().toString());
+        return createSession(params, UUID.randomUUID().toString(), ApplicationType.SESSION);
+    }
+
+    public Application createPermanentSession(String sessionId, SubmitParams params) {
+        return createSession(params, sessionId, ApplicationType.PERMANENT_SESSION);
     }
 
     public Application createSession(SubmitParams params, String sessionId) {
+        return createSession(params, sessionId, ApplicationType.SESSION);
+    }
+
+    public Application createSession(SubmitParams params, String sessionId, ApplicationType applicationType) {
```

Comment:
this should probably be private, otherwise we are able to create a new batch application through `createSession` method :)